bigquery/snippets/table/bigquery_table_exists.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package table
+
+// [START bigquery_table_exists]
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/api/googleapi"
+)
+
+// tableExists checks wheter a table exists on a given dataset.
+func tableExists(projectID, datasetID, tableID string) error {
+	// projectID := "my-project-id"
+	// datasetID := "mydatasetid"
+	// tableID := "mytableid"
+	ctx := context.Background()
+
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("bigquery.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	if _, err = tableRef.Metadata(ctx); err != nil {
+		if e, ok := err.(*googleapi.Error); ok {
+			if e.Code == http.StatusNotFound {
+				return errors.New("dataset or table not found")
```

Comment:
Yeah, I wish the backend was clearer about signalling the entity that isn't found, but alas.